auth/client/src/main/java/org/wildfly/security/auth/client/XMLParserUtils.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wildfly.security.auth.client;
+
+import org.wildfly.client.config.ConfigXMLParseException;
+import org.wildfly.client.config.ConfigurationXMLStreamReader;
+import org.wildfly.common.function.ExceptionSupplier;
+import org.wildfly.common.function.ExceptionUnaryOperator;
+
+import java.net.URI;
+
+import static org.wildfly.security.auth.client._private.ElytronMessages.xmlLog;
+
+/**
+ * An interface to get and check information about attributes in XML file.
```

Comment:
s/interface/utility class

Reply:
Since this is just a utility class with helper methods, this shouldn't be public API. Currently, this is a public class in a the org.wildfly.security.auth.client package and that is public API.

Reply:
Thanks for pointing out! 
Which of the following do you think would be the best approach: 
1. move it to org.wildfly.security.auth.util? 
2. keep it in the same directory and change the class and its functions to private or
3.  change the functions to protected and keep the class public or 
4. something else